Product Collection: Remove automatic migration code

WunderBart opened the pull request:
### Changes proposed in this Pull Request:

- Remove the code related to the automatic `Products (Beta)` -> `Product Collection` upgrade.
- Add an e2e test that ensures the manual upgrade works as expected.

Closes #42270

### How to test the changes in this Pull Request:

Ensure the manual upgrade works as expected:
1. Add the deprecated `Products (Beta)` block by pasting this snippet:

   ```html
   <!-- wp:query {"namespace":"woocommerce/product-query"} -->
   <div class="wp-block-query"></div>
   <!-- /wp:query -->
   ```
2. Open the (Block) Settings panel and click the `Upgrade to Product Collection`.
3. Ensure the block properly upgraded to Product Collection.

WunderBart committed:
Remove auto migration code

WunderBart committed:
Add E2E test for manual upgrade

WunderBart committed:
Merge branch 'trunk' into refactor/pc-remove-automatic-migration

WunderBart committed:
Add changelog entry

imanish003 reviewed (approved):
Hey @WunderBart, I reviewed the PR and it looks good to me. Also, Thanks for adding E2E test 🙌🏻

kmanijak reviewed (approved):
Looks good! 🚀 

I found one issue when testing manual upgrade which I fixed in this PR: https://github.com/woocommerce/woocommerce/pull/50613 (details inside). PR is pointing to this branch so @WunderBart I suggest merging https://github.com/woocommerce/woocommerce/pull/50613 to it and then merging yours.

kmanijak committed:
Add default value of empty array to selected products to prevent Handpicked Products control from crashing (#50613)

Add default value of empty array to selected products to prevent control from crashing
